src/ossf_scorecard/contrib/models.py

```diff
@@ -0,0 +1,119 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Visit https://github.com/nexB/ScoreCode for support and
+# download.
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+class Package_score_Mixin(models.Model):
+    """
+    Abstract Model for saving OSSF scorecard data.
+    """
+
+    scoring_tool = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text=_(
+            "Defines the source of a score or any other scoring metrics"
+            "For example: ossf-scorecard for scorecard data"
+        ),
+    )
+
+    scoring_tool_version = models.CharField(
+        max_length=50,
+        blank=True,
+        help_text=_(
+            "Defines the version of the scoring tool used for scanning the"
+            "package"
+            "For Eg : 4.6 current version of OSSF - scorecard"
+        ),
+    )
+
+    score = models.CharField(
+        max_length=50,
+        blank=True,
+        help_text=_("Score of the package which is scanned"),
+    )
+
+    scoring_tool_documentation_url = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text=_("Documentation URL of the scoring tool used"),
+    )
+
+    score_date = models.DateTimeField(
+        blank=True,
+        null=True,
+        editable=False,
+        help_text=_(
+            "Date when the scoring was calculated on the package"
+        ),
+    )
+
+    class Meta:
+        abstract = True
+
+
+class scorecard_checks_Mixin(models.Model):
+
+    check_name = models.CharField(
+        max_length=100,
+        blank=True,
+        help_text=_(
+            "Defines the name of check corresponding to the OSSF score"
+            "For example: Code-Review or CII-Best-Practices"
+            "These are the some of the checks which are performed on a scanned package"
+        ),
+    )
+
+    check_score = models.CharField(
+        max_length=50,
+        blank=True,
+        help_text=_(
+            "Defines the score of the check for the package scanned"
+            "For Eg : 9 is a score given for Code-Review"
+        ),
+    )
+
+    reason = models.CharField(
+        max_length=300,
+        blank=True,
+        help_text=_("Gives a reason why a score was given for a specific check"
+                    "For eg, : Found 9/10 approved changesets -- score normalized to 9"),
+    )
+
+    details = models.JSONField(
+        default=list,
+        blank=True,
+        help_text=_(
+            "A list of details/errors reharding the score"
+        ),
+    )
+
```

Comment:
See https://github.com/nexB/ScoreCode/issues/3 we need two things:

1. django mixin models which are not used/populated here
2. normal python classes (maybe with attrs) which store the data here at `src/ossf_scorecard/scorecard.py` and returns this. This has to be used and tested.

Reply:
Hi @AyanSinhaMahapatra I could not understand this part.

Python classes to store the data in the model; should we keep that code in scorecode.

Reply:
> Python classes to store the data in the model; should we keep that code in scorecode.

Yes

Reply:
Please refer to the below link for the object models

https://github.com/nexB/ScoreCode/pull/5/files#diff-56f1d3090dbf7f6d6bf0d756e8ecdbdfe4697289119d8e98e3ce99121e197041